tests/services/logging.service.spec.ts

```diff
@@ -1,64 +1,160 @@
-const oldLogIdentifier = 'Old Log';
-const repeatOldLogStartHeader: (numberOfRepitions: number) => string = (numberofRepitions: number) => {
-  return `${loggingService.LOG_START_HEADER}\n${oldLogIdentifier}\n`.repeat(numberofRepitions);
+let headerLog: string;
+const sessionSeparator = '\n\n';
+const mockDate = new Date(2021, 6, 27);
+const infoLogMessage = 'Info log message';
+const errorLogMessage = 'Error log message';
+const warnLogMessage = 'Warn log message';
+const debugLogMessage = 'Debug log Message';
+
+const testAddLog = (message: string) => {
+  loggingService.startSession();
+  const initialLog = loggingService.getCachedLog();
+  loggingService.info(message);
+  const actualLog = loggingService.getCachedLog();
+  const expectedLog = `${initialLog}\n${message}`;
+  expect(actualLog).toEqual(expectedLog);
 };
-const logHeaderFilter: (line: string) => boolean = (line: string) => line === loggingService.LOG_START_HEADER;
-const oldLogFilter: (line: string) => boolean = (line: string) => line === oldLogIdentifier;
 
 describe('LoggingService', () => {
-  beforeAll(() => {
+  beforeEach(() => {
+    // Initialize mock local storage
+    let mockLocalStorage = {};
+    const mockLocalStorageFunctions = {
+      getItem: (key: string): string => {
+        return key in mockLocalStorage ? mockLocalStorage[key] : null;
+      },
+      setItem: (key: string, value: string) => {
+        mockLocalStorage[key] = value;
+      },
+      removeItem: (key: string) => {
+        delete mockLocalStorage[key];
+      },
+      clear: () => {
+        mockLocalStorage = {};
+      }
+    };
+
+    // Mock function calls
+    spyOn(localStorage, 'getItem').and.callFake(mockLocalStorageFunctions.getItem);
+    spyOn(localStorage, 'setItem').and.callFake(mockLocalStorageFunctions.setItem);
+    spyOn(localStorage, 'removeItem').and.callFake(mockLocalStorageFunctions.removeItem);
+    spyOn(localStorage, 'clear').and.callFake(mockLocalStorageFunctions.clear);
+
+    // Mock dates
+    jasmine.clock().install();
+    jasmine.clock().mockDate(mockDate);
+
+    // Initialize logging service
     const electronService = jasmine.createSpyObj('ElectronService', ['isElectron']);
     electronService.isElectron = jasmine.createSpy('isElectron', () => false);
     loggingService = new LoggingService(electronService);
+    headerLog = `${loggingService.LOG_START_HEADER}\n${mockDate.toLocaleString()}`;
+    loggingService.reset();
+    localStorage.clear();
   });
 
-  describe('.getTrimmedLogCache()', () => {
-    it('should return 1 new log if cache does not contain existing log', () => {
-      expect(getFilteredLogCount(undefined, logHeaderFilter)).toEqual(1);
-      expect(getFilteredLogCount('', logHeaderFilter)).toEqual(1);
-      expect(getFilteredLogCount('gibberish', logHeaderFilter)).toEqual(1);
+  afterEach(() => {
+    loggingService.reset();
+    localStorage.clear();
+    jasmine.clock().uninstall();
```

Comment:
Turns out that we need to cleanup (uninstall jasmine clock) in `afterAll`, otherwise some other tests in other files would break due to timeout errors (due to the hardcoded clock)